### DISCUSSION TITLE
[Chapter 6] - Q & A (1부)

### DISCUSSION THREAD

**Discussion by Irisation23** [Study-2-Modern-Java-In-Action/Study-2-Modern-Java-in-Action#60] [CHAPTER 6 스트림으로 데이터 수집 - Q&A]
책의 내용중에 자신의 상황에 맞는 최적의 해법 선택을 하라고 하였다.

지금까지 책에서는 컬렉터를 이용하는 것 보단 스트림 메서드를 이용하는 것이 코드 복잡도를 줄여주는 장점이 있었는데 컬렉터를 사용할 때 코드가 복잡한 대신 재사용성과 커스터마이징 가능성을 제공하는 높은 수준의 추상화와 일반화를 얻을 수 있다고 하였다. 

### Q. 스트림 메서드를 사용하면서도 높은 추상화, 일반화를 얻을 수 없을까?

**Comment by corock** [comment]
스트림 메서드와 컬렉터의 장단점을 비교할 만한 예시가 떠오르지 않네요...
스트림도 인터페이스이기 때문에 필요한 경우라면 `implement` 해서 커스터마이징은 해볼 수 있을 것 같습니다.
